core/validation/initial_check.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any
+
+import numpy as np
+import pandas as pd
+
+import core.validation.failures as vf
+from core.const import EXPECTED_ROUND_THREE_SHEETS
+from core.validation.failures import NoInputFailure
+
+
+def extract_round_three_submission_details(
+    workbook: dict[str, pd.DataFrame]
+) -> dict[str, list[str]] | dict[str, dict[Any, Any]]:
+    """
+    Extract submission details for round three from the given workbook.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas DataFrames.
+    :return: A dictionary containing the extracted submission details.
+    """
+    details_dict = {"ValueChecks": {}, "NullChecks": {}}
+
+    missing_sheets = check_missing_sheets(EXPECTED_ROUND_THREE_SHEETS, workbook)
+
+    if missing_sheets:
+        return {"Missing Sheets": missing_sheets}
+
+    sheet_a1 = workbook.get("1 - Start Here")
+    if sheet_a1 is not None:
+        details_dict["ValueChecks"]["Form Version"] = (
+            sheet_a1.iloc[6][1],
+            {"Town Deals and Future High Streets Fund Reporting Template (v3.0)"},
+        )
+        details_dict["ValueChecks"]["Reporting Period"] = (sheet_a1.iloc[4][1], {"1 October 2022 to 31 March 2023"})
+
+    sheet_a2 = workbook.get("2 - Project Admin")
+    if sheet_a2 is not None:
+        details_dict["ValueChecks"]["Fund Type"] = (sheet_a2.iloc[5][4], {"Town_Deal", "Future_High_Street_Fund"})
+        details_dict["NullChecks"]["Place Name"] = sheet_a2.iloc[6][4]
+
+    return details_dict
+
+
+def pre_transformation_check(submission_details: dict[str, list[str]]) -> list[vf.ValidationFailure]:
+    """
+    Perform pre-transformation checks on the given workbook.
+
+    :param submission_details: A dictionary containing submission details.
+    :return: A list of validation failures encountered during pre-transformation checks.
+    """
+
+    if missing_sheets := submission_details.get("Missing Sheets"):
+        return [vf.EmptySheetFailure(empty_sheet) for empty_sheet in missing_sheets]
+
+    failures = []
+
+    for value_descriptor, (entered_value, expected_values) in submission_details["ValueChecks"].items():
+        if failure := check_values(value_descriptor, entered_value, expected_values):
+            failures.append(failure)
+
+    for value_descriptor, values in submission_details["NullChecks"].items():
+        if failure := check_nulls(value_descriptor, values):
+            failures.append(failure)
+
+    return failures
+
+
+def check_values(
+    value_descriptor: str, entered_value: str, expected_values: tuple
```

Comment:
Looks like `expected_values` is a set rather than a tuple

Reply:
docstring needs updating to reflect

Reply:
Amended.